### ISSUE TITLE
quickstart wizard should render the pipeline choices using a visual pipeline representation

### ISSUE THREAD

**Issue by jstrachan** [fabric8io/fabric8-ui#1043]
rather than just a list of pipeline names it'd be nice to show a list of pipeline renderings for each pipeline choice.

The JSON returned by the wizard has the stages, environments and a description.

e.g.
```json
    "inputs":
    [
        {
            "name":"pipeline",
            "shortName":" ",
            "valueType":"io.fabric8.forge.generator.pipeline.PipelineDTO",
            "inputType":"org.jboss.forge.inputType.DEFAULT",
            "enabled":true,
            "required":false,
            "deprecated":false,
            "label":"Pipeline",
            "description":"The Jenkinsfile used to define the Continous Delivery pipeline",
            "valueChoices":
            [
                {
                    "id":"Release",
                    "builder":"maven",
                    "descriptionMarkdown":"Maven based pipeline which:\n\n* creates a new version then builds and deploys the project into the maven repository\n* runs an integration test in the **Test** environment\n* stages the new version into the **Stage** environment\n",
                    "environments":
                    [
                        "Test"
                    ],
                    "label":"Release",
                    "stages":
                    [
                        "Build Release",
                        "Integration Test"
                    ],
                    "value":"maven/Release/Jenkinsfile"
                },
                {
                    "id":"ReleaseAndStage",
                    "builder":"maven",
                    "descriptionMarkdown":"Maven based pipeline which:\n\n* creates a new version then builds and deploys the project into the maven repository\n* runs an integration test in the **Test** environment\n* stages the new version into the **Stage** environment\n",
                    "environments":
                    [
                        "Test",
                        "Stage"
                    ],
                    "label":"ReleaseAndStage",
                    "stages":
                    [
                        "Build Release",
                        "Integration Test",
                        "Rollout to Stage"
                    ],
                    "value":"maven/ReleaseAndStage/Jenkinsfile"
                },
                {
                    "id":"ReleaseStageApproveAndPromote",
                    "builder":"maven",
                    "descriptionMarkdown":"Maven based pipeline which:\n\n* creates a new version then builds and deploys the project into the maven repository\n* runs an integration test in the **Test** environment\n* stages the new version into the **Stage** environment\n* waits for **Approval** to promote \n* promotes to the **Run** environment\n",
                    "environments":
                    [
                        "Test",
                        "Stage",
                        "Run"
                    ],
                    "label":"ReleaseStageApproveAndPromote",
                    "stages":
                    [
                        "Build Release",
                        "Integration Test",
                        "Rollout to Stage",
                        "Approve",
                        "Rollout to Run"
                    ],
                    "value":"maven/ReleaseStageApproveAndPromote/Jenkinsfile"
                }
            ],
            "class":"UISelectOne",
            "value":"ReleaseStageApproveAndPromote"
        }
    ]
```

For example the `ReleaseStageApproveAndPromote` pipeline choice has stages:

* Build Release
* Integration Test
* Rollout to Stage
* Approve
* Rollout to Run

To render that like the pipelines UI it should be fairly easy - something along these lines

```html
<div class="pipeline">
  <div class="pipeline-stage" **ngFor="let stage of stages">
    <div class="pipeline-stage-column">
      <div class="pipeline-stage-name Cancelled" title="stage status: SUCCESS">{{stage}}</div>
        <div class="pipeline-status-bar SUCCESS">
          <div class="pipeline-line"></div>
          <div class="pipeline-circle">
            <div class="clip1"></div>
            <div class="clip2"></div>
            <div class="inner-circle">
              <div class="inner-circle-fill"></div>
            </div>
          </div>
        </div>
    </div>
  </div>
</div>
```

then we'd just need similar CSS loaded like is used in the fabric8-runtime-console's pipeline rendering:
https://github.com/fabric8-ui/fabric8-runtime-console/blob/master/src/app/kubernetes/ui/pipeline/build-stage-view/build-stage-view.component.scss

(we could omit a fair bit of that CSS as we'd only show SUCCESS states and would not show animations)

**Comment by jstrachan** [comment]
here's an example of the UI in action for viewing a pipeline - we'd not have any details of the timings below the lines/arrows/circles though

<img width="1193" alt="openshift_io" src="https://cloud.githubusercontent.com/assets/30140/25382807/14351f9c-29b1-11e7-8be3-a2ecc6df5586.png">

**Comment by joshuawilson** [comment]
@jstrachan where in the wizard are the pipelines shown?

**Comment by jstrachan** [comment]
on the second page of the wizard when you choose a pipeline

**Comment by jstrachan** [comment]
here's the second page of the new quickstart wizard:
<img width="1005" alt="fabric8" src="https://cloud.githubusercontent.com/assets/30140/25399915/e0605596-29e8-11e7-9699-a94e87987d59.png">

**Comment by joshuawilson** [comment]
@jstrachan so are you asking that an image of the pipeline be shown for each selection or something else. 
cc @catrobson

**Comment by pmuir** [comment]
Yes, we need to do this. If we can render dynamically, even better.

**Comment by jstrachan** [comment]
I remember @pmuir asking for this recently. 

Basically rather than showing a combo list of pipeline names (with just a textual description of the pipeline), the idea is to show a list of pipelines using the same visualisation as the pipeline visualisation above.

e.g. show the, currently 3 different pipelines and their stages and let the user pick which one they want

**Comment by jstrachan** [comment]
We've explicitly added the 'Stages' metadata to all the pipelines in the library - solely so that UIs can render the stages in each kind of pipeline to help the user pick which ones to use. It turns out that little bit of code above is pretty much all we need to do to render a static pipeline flow; using the list of stages from each pipeline object the server returns.

**Comment by jstrachan** [comment]
@pmuir I was thinking just a static rendering of each pipeline choice (making it look like they all completed with green)? Though I guess we could animate them but not sure that extra work is really needed?

**Comment by pmuir** [comment]
The only thing I'm not sure that shows well is the approval step...

**Comment by jstrachan** [comment]
Yeah - for approval stages we could use the `PAUSED_PENDING_INPUT` css style instead of `SUCCESS` to highlight input is required; though when the pipeline is complete if `Promote` is selected it goes green with `SUCCESS` anyway so its maybe not a biggie for now?

Either way if we wanted to render `Approve` stages separately we can just check for the Stage name `Approve` for now - e.g. https://github.com/fabric8io/fabric8-jenkinsfile-library/blob/master/maven/ReleaseStageApproveAndPromote/metadata.yml#L9

**Comment by catrobson** [comment]
We had an original design that did not show the pipeline visualization, but rather listed a summary and extended to show full stage listing.  Is this a reasonable place to start?  https://redhat.invisionapp.com/share/ZQARDX35J#/screens/228077943

![image](https://cloud.githubusercontent.com/assets/3739227/25412412/7fbf4834-29f0-11e7-8caf-8915615553b0.png)

Ideally I'd like to create a micro-pipeline (use on dashboards), mini-pipeline (use for instances like this), and the full-pipeline visualizations (full pipeline page), but this will require some visual design cycles to achieve so that we have a clear understanding of "proposed pipeline" states vs. "active pipeline" states for the visualization.

**Comment by jstrachan** [comment]
Ah I'd not seen that - its pretty nice yeah. Though do think just rendering what the pipeline looks like might be easier to grok for folks; particularly if they've seen fabric8/openshift/jenkins before - not sure folks understand 'stages' and 'environments' or if the number of each helps (in the un-expanded view); there's often confusion there what those words mean.

I wonder if the above UI might look simpler if its just the first column (`Build Image, Canary Release and Stage, ...`) then if folks expand it shows the pipeline visualisation?

We currently only have about 2 or 3 pipelines to choose from and I'm not sure the list is gonna grow much over time really so I figured we'd default to expand each row anyway as the this page of the UI is literally just picking from a list of 2-3 things.

Incidentally I just updated the forge server to render the pipelines as `Release and Stage` or `Release, Stage, Approve and Promote` etc. I wonder if I should remove the commas?
